### PR TITLE
Add DS2431 EEPROM emulation

### PR DESCRIPTION
**Pull request by j-langlois**
* DS2431 Emulation
* Added an optional CRC_INIT value in the crc16 calculation 
* 0x33 implementation since my masters use a single device on the bus

### CODE REVIEW

**Comment by orgua** [comment]
this looks awesome. Clean and simple extension of the hub-base.
i will merge it now and test it later.

File: src/DS2431.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// 0x2D  1Kb 1-Wire EEPROM
+// tested on buspirate and two different real 1-wire masters
```
Comment (orgua) [inline src/DS2431.h]: would be nice to hear about the masters. one seems to be the ds9490r. and the other? i plan on putting a compatibility-chart into the main readme.md

File: src/DS2431.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+#ifndef ONEWIRE_DS2431_H
+#define ONEWIRE_DS2431_H
+#define CRC_INIT 0xCFB5
```
Comment (orgua) [inline src/DS2431.h]: using a #define as include-guard is ok for now (i will switch to "#pragma once" later). 
the constant crc_init should be a constexpr. it has advantages like typesafety, no name collisions and more helpful compiler-debug-output

File: src/DS2431.h (hunk 1 above)
Comment (j-langlois) [inline src/DS2431.h]: It's a PIC18 MCU

File: src/DS2431.h (hunk 2 above)
Comment (mpsyco) [inline src/DS2431.h]: AFAIK, the `#pragma once` is not portable.

Comment (orgua) [inline src/DS2431.h]: what do you mean by that? yes it is not in the standard, but all except one c-compiler understand this pragma. Should be ok for quite a while now

Comment (mpsyco) [inline src/DS2431.h]: I meant that it is not in the standard, but I didn't knew about the wide-spread support for this (hence the "AFAIK").
Personnally, I prefer not using non-standard stuff, but [fair enough](http://stackoverflow.com/a/787539).

**Pull request merged**